### PR TITLE
kubeadm: Split out markmaster to its own phase

### LINKED ISSUE #53: kubeadm UpdateMasterRoleLabelsAndTaints should not use recursion
_From @errordeveloper on October 3, 2016 11:6_

_Copied from original issue: kubernetes/kubernetes#33913_

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:

Splits out related and atomic code into its own phase that can be invokable easily from the CLI.
Makes the code much easier to read by not using recursion and `wait.InfinitePoll` _inside_ of a recursing function, etc.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Fixes: https://github.com/kubernetes/kubeadm/issues/53
Part of this more long-term goal: https://github.com/kubernetes/kubeadm/issues/148

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```
@kubernetes/sig-cluster-lifecycle-pr-reviews

### CODE REVIEW

File: cmd/kubeadm/app/phases/markmaster/markmaster.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	clientset "k8s.io/client-go/kubernetes"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	kubeletapis "k8s.io/kubernetes/pkg/kubelet/apis"
+)
+
+// MarkMaster taints the master and sets the master label
+func MarkMaster(client clientset.Interface, masterName string) error {
+
+	// Loop on every falsy return. Return with an error if raised. Exit successfully if true is returned.
+	return wait.PollInfinite(kubeadmconstants.APICallRetryInterval, func() (bool, error) {
```
Comment (timothysc) [inline cmd/kubeadm/app/phases/markmaster/markmaster.go]: This is another wait forever experience, could you do a PollUntil with a timer which could be a const.

**Review by timothysc — DISMISSED** [review]
Minor comment to remove the forever polling.

File: cmd/kubeadm/app/phases/markmaster/markmaster.go (hunk 1 above)
Comment (luxas) [inline cmd/kubeadm/app/phases/markmaster/markmaster.go]: That was the case earlier, but makes sense to fixup now as well

**Comment by luxas** [comment]
Fixed the trivial minor nit and rebased, self-applying the lgtm label based on your review

**Pull request merged**